doc/guides/maintaining-the-build-files.md

```diff
@@ -0,0 +1,199 @@
+Note that the CI runs different targets on Raspberry Pis
+(`node-test-commit-arm-fanned`) and Linux with FIPS
+(`node-test-commit-linux-fips`).
+
+To update the process for most CI, update:
+
+- `run-ci`: does not affect the CI job on Raspberry Pis and Linux with FIPS.
+- `build-ci`: does not affect the CI job on Linux with FIPS.
+- `test-ci`: does not affect the CI job on Raspberry Pis
+
+To update the `node-test-linter` CI job, update:
+
+- `lint-ci`
+- `lint-js-ci`
+
+For the Raspberry Pis, update:
+
+- `test-ci-js`
+- `test-ci-native`
+
+#### `deps` Testing Targets 
+
+The normal testing process does not run tests of third-party libraries
+inside `deps`, because most of them can run their tests on their own. However,
+there are a few targets in the Makefile that are used to test V8 and npm,
+because we have our own floating patches (not merged in the upstream) that
+need to be tested. Consider updating these targets when you need to update
+the process of testing them:
+
+For `deps/v8`:
+
+- `v8`: rebuilds `deps/v8` as a git tree, pulls the third-party dependencies
+  and builds V8
+- `test-v8`: run by the `node-test-commit-v8-linux` CI job
+- `test-v8-benchmarks`
+- `test-v8-intl`
+- `test-v8-all`
+
+For `deps/npm`:
+
+- `test-npm`
+- `test-npm-publish`
+
+#### Other Test Targets
+
+The following targets use `node-gyp` bundled inside the npm in the codebase:
+
+- `test-addons`
+- `test-addons-napi`
+- `test-gc`
+
+There are many `test-*` targets that can be run to test a specific part of
+the codebase, most of them using `tools/test.py`. There is also:
+
+- `cctest`: run the C++ tests.
+
+#### Documentation Targets
+
+The Node.js API documentation is built within this codebase. Consider updating
+these targets when you update the build process of the documentation.
+
+- `doc`
+- `doc-only`: does not depend on a locally built Node.js binary
+- `test-doc`
+- `docclean`
+
+#### Lint Targets
+
+Consider updating these targets when you update the linting process.
+Note that these can not be run if the source code comes from
+the source tarball.
+
+- `lint`: runs linters on different type of files
+- `lint-js`: for linting JavaScript code.
+- `lint-cpp`: for linting C++ code.
+- `lint-md`: for linting the markdown documents.
+- `lint-clean`
+
+Also note that the node-test-linter CI job runs:
+
+- `lint-ci` instead of `lint`
+- `lint-js-ci` instead of `lint-js`
+
+#### Release Targets
+
+- `binary`: builds binary tarballs
+- `pkg`: builds MacOS installer
+- `tar`: builds source tarballs
+- `tar-headers`: builds header tarballs
+
+The following targets are strictly for the release builds on the
+release machines:
+
+- `binary-upload`
+- `doc-upload`
+- `pkg-upload`
+- `tar-upload`
+- `tar-headers-upload`
+
+### Environment Variables
+
+- `BUILDTYPE`
+- `BUILD_DOWNLOAD_FLAGS`
+- `BUILD_INTL_FLAGS`
+- `CONFIG_FLAGS`
+- `DESTDIR`
+- `DISTTYPE`
+- `FLAKY_TESTS`
+- `JOBS`
+- `LOGLEVEL`
+- `PREFIX`
+- `PYTHON`
+- `TEST_CI_ARGS`
```

Comment:
I'm not sure how useful this is without information about what the variables actually do, and that info would make more sense (to me) in a comment in the Makefile itself.